### PR TITLE
Bugfix #10603 [v100.1] The scroll in the History list doesn't behave correctly

### PR DESCRIPTION
**Pull request by yoanarios**
Issue #10603
 
- Update fetch logic and use the result from SearchTermGroupsManager.getSiteGroups to populate history with groups and individual sites.
- Bring back pagination to get more history results adding the option to reset when LibraryViewController is open to get latest results

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -968,6 +968,12 @@ class BrowserViewController: UIViewController {
             libraryViewController.selectedPanel = panel
         }
 
+        // Reset history panel pagination to get latest history visit
+        if let historyPanel = libraryViewController.viewModel.panelDescriptors.first(where: {$0.panelType == .history}),
+           let vcPanel = historyPanel.viewController as? HistoryPanelWithGroups {
+            vcPanel.viewModel.shouldResetHistory = true //resetHistory()
```
Comment (lmarceau) [inline Client/Frontend/Browser/BrowserViewController.swift:974]: nit: ` //resetHistory()` can be removed?

File: Client/Frontend/Library/LibraryPanels.swift [hunk 2]
```diff
@@ -94,7 +96,8 @@ class LibraryPanels: FeatureFlagsProtocol {
             tabManager: tabManager,
             imageName: "Bookmarks",
             accessibilityLabel: .LibraryPanelBookmarksAccessibilityLabel,
-            accessibilityIdentifier: "LibraryPanels.Bookmarks"),
+            accessibilityIdentifier: "LibraryPanels.Bookmarks",
```
Comment (lmarceau) [inline Client/Frontend/Library/LibraryPanels.swift:99]: nit: imageName should be in ImageIdentifiers, accessibilityIdentifier should be in AccessibilityIdentifier.swift

**Review by nishant2718 — APPROVED** [review]
A couple comments that @lmarceau IDed, and the rest is good!

**Comment by yoanarios** [comment]
> A couple comments that @lmarceau IDed, and the rest is good!

I'm working in the nits I will address them in a follow up PR

**Pull request merged**